vncviewer/Monitors.h

```diff
@@ -0,0 +1,140 @@
+/* Copyright 2021 Hugo Lundin <huglu@cendio.se> for Cendio AB
+ *
+ * This is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * This software is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this software; if not, write to the Free Software
+ * Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA  02111-1307,
+ * USA.
+ */
+
+#ifndef __MONITORS_H__
+#define __MONITORS_H__
+
+#include "MonitorArrangement.h"
+
+#include <vector>
+#include <set>
+#include <list>
+#include <map>
+
+#define DESCRIPTION_MAX_LEN 256
+
+enum ConfigurationMode {
+  CURRENT,
+  ALL,
+  SELECTED
+};
+
+class Monitors: public MonitorArrangementDelegate {
```

Comment:
Not a huge fan of this class. It seems to duplicate state, and take on multiple roles for various sections. It makes it difficult to follow how things are connected.

Have you looked at trying to split out what it does to the other components and try to make them more independent? E.g. avoid having a callback mechanism in the widget and instead have a classical `set()`/`get()` style API? And avoiding duplicating state in here by just giving the current string parameters to a helper function when it is time to calculate relevant monitors?

Reply:
The monitors class has been removed. It has been replaced with a more classical `set()`/`get()` style API on `MonitorArrangement`. Additionally, `MonitorIndicesParameter` (a subclass of `rfb::StringParameter` ) has  been added, which is responsible for mapping between our external index representation and the FLTK indices that is used internally. Overall, I would like to state that the structure is much more clear now.